executor/cluster_reader.go

```diff
@@ -95,7 +96,23 @@ func (e *clusterConfigRetriever) retrieve(ctx sessionctx.Context) ([][]types.Dat
 		rows [][]types.Datum
 		err  error
 	}
-	serversInfo, err := getClusterServerInfoWithFilter(ctx, e.extractor)
+	serversInfo, err := getClusterServerInfoWithFilter(ctx, e.extractor.NodeTypes, e.extractor.Addresses)
```

Comment:
If we filter server information at L99, then the servers injected by Fairpoint will not be filtered. I think it's ok that move failpoint logic to `getClusterServerInfoWithFilter` with an extra failpoint name.